Control coverage/Feature/VirtualMachineScaleSet.md

```diff
@@ -1,16 +1,729 @@
-| <b>ControlId:</b><br>Azure_VirtualMachineScaleSet_DP_Enable_Disk_Encryption<br><b>DisplayName:</b><br>Disk encryption should be applied on virtual machine scale sets<br><b>Description: </b><br> Disk encryption must be enabled on both OS and data disks for Windows Virtual Machine Scale Set |<b> ARM API to get configuration of a Virtual Machine Scale Set: </b> <br> /subscriptions/{subscriptionId}<br>/providers/Microsoft.Compute<br>/virtualMachineScaleSets<br>?api-version=2019-07-01<br><b>Properties:</b><br> properties.virtualMachineProfile<br>.extensionProfile.extensions[\*].type<br>properties.virtualMachineProfile<br>.extensionProfile.extensions.[\*].provisioningState <br><br><b>ARM API to get instance view of Virtual Machines <br>in a Virtual Machine Scale Set</b> <br> /subscriptions/{subscriptionId}<br>/resourceGroups/{resourceGroupName}<br>/providers/Microsoft.Compute<br>/virtualMachineScaleSets/{vmScaleSetName}<br>/virtualmachines/{instanceId}/instanceView<br>?api-version=2021-03-01 <br><b>Properties</b><br> disks.statuses[\*].code | <b>Passed: </b><br>Azure disk encryption extension is installed <br>and existing disks (OS and Data) are encrypted.<br><b>Failed: </b><br>Azure disk encryption extension is not installed<br>One or more existing disks (OS or Data) are in a non-compliant state <br>(Any other state other than Encrypted is a non-compliant state. <br>Possible states: Encrypted/NotEncrypted/NotMounted/DecryptionInProgress<br>/EncryptionInProgress/VMRestartPending/Unknown/NoDiskFound) <br><b>Verify: </b><br>Storage profile for VM Scale Set unavailable or number of <br>Virtual Machine instances in VM Scale Set exceeded scan limit. |
+# VirtualMachineScaleSet
 
+**Resource Type:** Microsoft.Compute/virtualMachineScaleSets
+
+___
+
+## Azure_VirtualMachineScaleSet_Audit_Enable_Diagnostics
+
+### DisplayName
+Diagnostics (IaaSDiagnostics extension on Windows; LinuxDiagnostic extension on Linux) must be enabled on Virtual Machine Scale Set.
+
+### Rationale
+Diagnostics logs are needed for creating activity trail while investigating an incident or a compromise.
+
+### Control Settings
+```json
+{
+    "LinuxExtensionType": "LinuxDiagnostic",
+    "LinuxExtensionPublisher": "Microsoft.OSTCExtensions",
+    "WindowsExtensionType": "IaaSDiagnostics",
+    "WindowsExtensionPublisher": "Microsoft.Azure.Diagnostics"
+}
+```
+
+### Control Spec
+
+> **Passed:**
+> Required diagnostics extension is present in VM Scale Set.
+>
+> **Failed:**
+> Required diagnostics extension is missing in VM Scale Set.
+>
+> **Verify:**
+> Not Applicable.
+>
+> **NotApplicable:**
+> Not Applicable.
+>
+### Recommendation
+
+<!--
+- **Azure Portal**
+-->
+
+-**PowerShell**
+
+    ```powershell
```

Comment:
Formatting does not come right. Pls check

Reply:
Resolved